File: app/src/androidTest/java/eu/sesma/dagger/platform/TestAndroidApplication.java
package eu.sesma.dagger.platform;

import eu.sesma.dagger.di.ApplicationComponent;
import eu.sesma.dagger.di.DaggerTestApplicationComponent;

public class TestAndroidApplication extends AndroidApplication {
    @Override
    protected ApplicationComponent createComponent() {
        return DaggerTestApplicationComponent.builder()
                .testApplicationModule(new TestApplicationModule(this))
                .build();
    }
}

File: app/src/androidTest/java/eu/sesma/dagger/platform/TestAndroidApplication.kt
package eu.sesma.dagger.platform

import eu.sesma.dagger.AndroidApplication
import eu.sesma.dagger.di.ApplicationComponent
import eu.sesma.dagger.di.DaggerTestApplicationComponent

class TestAndroidApplication : AndroidApplication() {
    override fun createComponent(): ApplicationComponent {
        return DaggerTestApplicationComponent.builder()
                .testApplicationModule(TestApplicationModule(this))
                .build()
    }
}
